src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkAddressChange.Windows.cs

```diff
@@ -57,28 +57,35 @@ private static void RunHandlerCallback(object state)
 
             private static void ChangedAddress(object sender, EventArgs eventArgs)
             {
+                Dictionary<NetworkAvailabilityChangedEventHandler, ExecutionContext> s_copy = null;
+
                 lock (s_globalLock)
                 {
                     bool isAvailableNow = SystemNetworkInterface.InternalGetIsNetworkAvailable();
 
+                    // If there is an Availability Change, need to execute user callbacks.
                     if (isAvailableNow != s_isAvailable)
                     {
                         s_isAvailable = isAvailableNow;
 
-                        var s_copy =
+                        s_copy =
                             new Dictionary<NetworkAvailabilityChangedEventHandler, ExecutionContext>(s_availabilityCallerArray);
+                    }
+                }
 
-                        foreach (var handler in s_copy.Keys)
+                // Executing user callbacks if Availability Change event occured.
+                if (s_copy != null)
+                {
+                    foreach (var handler in s_copy.Keys)
```

Comment:
Can this just enumerate s_copy instead of enumerating s_copy.Keys?  The body can then use entry.Key to get the handler.  That avoids needing to allocate the KeysCollection returned from Keys.

Reply:
Will fix. I will also change the naming. (s_copy -> copy)